runtime/vm/visible.c

```diff
@@ -100,12 +105,21 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 			}
 		} else if (modifiers & J9AccPrivate) {
 			/* Private */
-			if ((sourceClass != destClass)
+			if (sourceClass != destClass) {
 #if defined(J9VM_OPT_VALHALLA_NESTMATES)
-					&& (sourceClass->nestHost != destClass->nestHost)
+				/* loadAndVerifyNestHost returns an error if setting nest host field fails */
+				if (NULL == destClass->nestHost) {
+					result = loadAndVerifyNestHost(currentThread, destClass);
```

Comment:
if we add a `goto`, we can avoid the repeated `(J9_VISIBILITY_ALLOWED == result)` check in the common path.

Something like:
```
if (NULL == destClass->nestHost) {
	result = loadAndVerifyNestHost(currentThread, destClass);
	if ((J9_VISIBILITY_ALLOWED != result) {
		goto _exit;
	}
}
if (NULL == sourceClass->nestHost) {
....
}
```